storage/innobase/btr/btr0btr.cc

```diff
@@ -921,17 +921,46 @@ btr_page_get_father(
 	mem_heap_free(heap);
 }
 
+/** Check whether AHI is enabled or not. Not this is only applicable
+for dropping of indexes during truncate.
+@param[in]	is_truncate 	true for truncate operation
+@return true always for non-truncate operations, for truncate, check
+the AHI status under lock */
+static
+bool
+btr_check_ahi_status(bool is_truncate) {
+
+	if (!is_truncate) {
+		return(true);
+	}
+
+	/* At drop index stage (only for truncate), concurrently AHI can only
+	be enabled. It cannot be disabled because disabling needs dict_sys
+	mutex. See btr_search_disable(). We already own dict_sys mutex here.
+	Lock and get status of AHI. If we read it as disabled under
+	lock, it can become enabled after lock release. But it doesn't
+	matter because no one can add AHI entries at this stage. So
+	what we read under lock can be trusted */
+	ut_ad(mutex_own(&dict_sys->mutex));
+	btr_search_s_lock_all();
```

Comment:
Don't need to lock all: any single partition should be enough

Reply:
True, it can be improved. index_id should be passed from different paths. Since it is only about reading status, this should be quick.

Reply:
You don't even need index_id. [0] or any other ID will work.

Reply:
Hmm, since it only about reading the variable under lock, I guess there is no harm in using 'any' index_id.

Will try to implement this

Reply:
done